app/controllers/api/v2/instance_hosts_controller.rb

```diff
@@ -0,0 +1,34 @@
+module Api
+  module V2
+    class InstanceHostsController < V2::BaseController
+      def resource_class
+        @resource_class ||= Host::Managed
+      end
+
+      def resource_scope(*args)
+        super.authorized(:view_hosts).joins(:infrastructure_facet).merge(::HostFacets::InfrastructureFacet.where(foreman: true))
+      end
+
+      api :PUT, '/instance/hosts/:host_id', N_("Assign a host to the Foreman instance")
+      param :host_id, :identifier_dottable
+      def update
+        # We cannot use resource scope as that is scoped only to hosts which already have the facet
+        host = ::Host::Managed.friendly.find(params[:id])
```

Comment:
Help me understand this since it's new to me. Why isn't this using `resource_class`?
```suggestion
        host = resource_class.friendly.find(params[:id])
```

Also, why is this `::Host` while it's `Host` in `resource_class`?

Reply:
> Why isn't this using resource_class?

Most likely I wrote this part before overriding `resource_class`.

> Also, why is this ::Host while it's Host in resource_class?

No particular reason. Both should resolve to the same thing.